helper/schema/set.go

```diff
@@ -56,6 +56,19 @@ func (s *Set) List() []interface{} {
 	return result
 }
 
+// MapByKey returns the elements of this set as map with defined key
+// 
+// Duplicate elements will be eliminated (last win)
+func (s *Set) MapByKey(key string) map[string]interface{} {
```

Comment:
I think we should just move this out into a helper in the AWS provider. I don't see this as being generally useful enough to put onto the `Set` directly.